packages/material-ui/src/LinearProgress/LinearProgress.js

```diff
@@ -248,6 +255,7 @@ LinearProgress.propTypes = {
 LinearProgress.defaultProps = {
   color: 'primary',
   variant: 'indeterminate',
+  styles: {},
```

Comment:
I would do the following:
```ts
export type LinearProgressStyleKeys =
  | 'bar1'
  | 'bar2';
```
and then, `style: {}` can be changed to
```ts
style: { [P in LinearProgressStyleKeys]?: React.CSSProperties };
```

For now, regarding the type definition, this can be seen as a temporary fix. In the long run, we should introduce the `styles` api and standardize it just like `XXXClassKey` type. (Probably add `XXXStylesKey` as the third generic type parameter to `StandardProps`? like `StandardProps<React.HTMLAttributes<HTMLDivElement>, LinearProgressClassKey, LinearProgressStylesKey>` in this case. @pelotom what do you think?)